File: 3.go
package main

import (
	"fmt"
	"os"
)

// Функция для переворота числа и удаления ведущих нулей
func revers_num(num int) int {
	revers := 0
	for num != 0 { // Переворачиваем число
		revers = revers*10 + num%10
		num /= 10
	}
	return revers
}

func main() {
	var n int
	_, err := fmt.Scan(&n)
	// Проверка на корректность ввода
	if err != nil || n <= 0 {
		fmt.Println("Количество элементов должно быть положительным и целым")
		os.Exit(1) // Завершаем программу с ошибкой если число не положительное или нецелое
	} else {
		for i := 0; i < n; i++ {
			var num int
			_, err := fmt.Scan(&num)
			// Проверка на корректность ввода
			if err != nil || num <= 0 {
				fmt.Println("Числа должны быть положительным и целым")
				os.Exit(1) // Завершаем программу с ошибкой если число не положительное или нецелое
			} else {
				fmt.Printf("%d ", revers_num(num)) // Вызываем функцию revers_num и выводим
			}
		}
	}

	os.Exit(0)
}

File: 3.rs
use std::io;
use std::process;

// Функция для переворота числа и удаления ведущих нулей
fn revers_num(mut num: i32) -> i32 {
    let mut revers = 0;
    while num != 0 {  // Переворачиваем число
        revers = revers * 10 + num % 10;
        num /= 10;
    }
    revers
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Ошибка чтения");
    let n: i32 = match input.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Количество элементов должно быть положительным и целым");
            process::exit(1);
        }
    };

    // Проверка на корректность ввода
    if n <= 0 {
        println!("Количество элементов должно быть положительным и целым");
        process::exit(1);
    } else {
        for _ in 0..n {
            let mut num_input = String::new();
            io::stdin().read_line(&mut num_input).expect("Ошибка чтения");
            let num: i32 = match num_input.trim().parse() {
                Ok(num) => num,
                Err(_) => {
                    println!("Числа должны быть положительным и целым");
                    process::exit(1);
                }
            };

            // Проверка на корректность ввода
            if num <= 0 {
                println!("Числа должны быть положительным и целым");
                process::exit(1);
            } else {
                print!("{} ", revers_num(num));  // Вызываем функцию revers_num и выводим
            }
        }
    }

    process::exit(0);
} 
